### ISSUE TITLE
Cursor in first position of URL field after entering text and autocompleted URL restored in field

### ISSUE THREAD

**Issue by AaronMT** [mozilla-mobile/firefox-tv#726]
- Click the URL field on the home-screen
- Clear the field and enter 'a' to autocomplete to amazon.com 
- Hit back 

Expected to see the cursor after the 'a' in the URL field

Second issue, while viewing the above issue:

- Hit settings
- Go back

See the URL field autocompleted magically to 'amazon.com' with the cursor in position 0 and mazon.com highlighted.

![ezgif-2-2a06d1a290](https://user-images.githubusercontent.com/102331/37916706-8a87c4ec-30ea-11e8-97b9-1aca726041a8.gif)

**Comment by Sdaswani** [comment]
@lime124 can you comment how annoying this for users?

**Comment by Sdaswani** [comment]
Also may be changing due to home screen work #690

**Comment by mcomella** [comment]
On the overlay, we only let users focus around the URL bar, not actually add a cursor into the field itself: this solves that problem there and we could apply that solution here.

**Comment by lime124** [comment]
yes - this is something that we'll keep in mind and address while taking a look at the homescreen designs.

let's leave this issue open for now and we can either use this to just fix the search bar for the release or have another issue that covers this and other homescreen revisions we want to make for the release.
